Wangscape/TilePartitionerBase.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <vector>
+#include <SFML/Graphics.hpp>
+#include <functional>
+#include "Options.h"
+
+class TilePartitionerBase
+{
+public:
+    typedef std::vector<std::pair<sf::Texture, TerrainID>> TilePartition;
+    typedef std::vector<TerrainID> Corners;
```

Comment:
Good idea. The codebase is already large enough that applying a directory structure would be appropriate, and many of the typedefs don't really belong inside a class.